src/plans/D3MAaveDaiPlan.sol

```diff
@@ -0,0 +1,167 @@
+        require((z = x - y) <= x, "D3MAaveDaiPlan/underflow");
+    }
+    function _mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x, "D3MAaveDaiPlan/overflow");
+    }
+    function _rmul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, y) / RAY;
+    }
+    function _rdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, RAY) / y;
+    }
+
+    // --- Admin ---
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "bar") {
+            require(data <= maxBar(), "D3MAaveDaiPlan/above-max-interest");
+
+            bar = data;
+        } else revert("D3MAaveDaiPlan/file-unrecognized-param");
+    }
+
+    function maxBar() public view returns (uint256) {
+        return interestStrategy.getMaxVariableBorrowRate();
+    }
+
+    // --- Automated Rate targeting ---
+    function calculateTargetSupply(uint256 targetInterestRate) external view returns (uint256) {
+        uint256 stableDebtTotal = stableDebt.totalSupply();
+        uint256 variableDebtTotal = variableDebt.totalSupply();
+        return _calculateTargetSupply(targetInterestRate, stableDebtTotal, variableDebtTotal);
+    }
+
+    function _calculateTargetSupply(uint256 targetInterestRate, uint256 stableDebtTotal, uint256 variableDebtTotal) internal view returns (uint256) {
+        uint256 base = interestStrategy.baseVariableBorrowRate();
+        require(targetInterestRate > base, "DssDirectDepositAaveDai/target-interest-base");
+        require(targetInterestRate <= interestStrategy.getMaxVariableBorrowRate(), "DssDirectDepositAaveDai/above-max-interest");
+
+        // Do inverse calculation of interestStrategy
+        uint256 variableRateSlope1 = interestStrategy.variableRateSlope1();
+        uint256 targetUtil;
+        if (targetInterestRate > _add(base, variableRateSlope1)) {
+            // Excess interest rate
+            uint256 r = targetInterestRate - base - variableRateSlope1;
+            targetUtil = _add(_rdiv(_rmul(interestStrategy.EXCESS_UTILIZATION_RATE(), r), interestStrategy.variableRateSlope2()), interestStrategy.OPTIMAL_UTILIZATION_RATE());
+        } else {
+            // Optimal interest rate
+            targetUtil = _rdiv(_rmul(_sub(targetInterestRate, base), interestStrategy.OPTIMAL_UTILIZATION_RATE()), variableRateSlope1);
+        }
+        return _rdiv(_add(stableDebtTotal, variableDebtTotal), targetUtil);
+    }
+
+    function getTargetAssets(uint256 currentAssets) external override view returns (uint256) {
+        uint256 targetInterestRate = bar;
+        if (targetInterestRate == 0) return 0;     // De-activated
+
+        uint256 totalPoolSize = _add(
+                TokenLike(dai).balanceOf(address(adai)),
+                _add(
+                    stableDebt.totalSupply(),
+                    variableDebt.totalSupply()
+                )
```

Comment:
This `stableDebt + variableDebt` addition is re-used in several places.

I think you can reduce all these references to

```
uint256 totalDebt = _add(
    stableDebt.totalSupply(),
    variableDebt.totalSupply()
);
```

and just pass that reference around. This will get rid of a duplicate lookup below as well.